Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -947,17 +947,26 @@ protected IEnumerable<DataGridColumn<TItem>> DisplayableColumns
         /// <summary>
         /// Returns true if EmptyTemplate is set and Data is null or empty.
         /// </summary>
-        protected bool IsEmptyTemplateVisible => !IsLoadingTemplateVisible && !IsNewItemInGrid && EmptyTemplate != null && ( Data == null || !Data.Any() );
+        protected bool IsEmptyTemplateVisible
+            => !IsLoadingTemplateVisible && !IsNewItemInGrid && EmptyTemplate != null && ( Data == null || !Data.Any() );
+
+        /// <summary>
+        /// Returns true if EmptyFilterTemplate is set and FilteredData is null or empty.
+        /// </summary>
+        protected bool IsEmptyFilterTemplateVisible
+            => !IsLoadingTemplateVisible && !IsNewItemInGrid && EmptyFilterTemplate != null && ( !( Data == null || !Data.Any() ) && ( FilteredData == null || !FilteredData.Any() ) );

```

Comment:
I don't have time to go test this now. But does this work ok with ReadData?
Can I not have 
**Data empty or null** + FilteredData Empty or null 
and it being a valid case for Empty Filter?

Reply:
It works with ReadData also. I specifically separated them because otherwise they could both be shown at the same time.